Adding Fedora CI docker image.

cathay4t opened the pull request:
Fedora CI image has been uploaded to docker hub as `nmstate/fedora-nmstate-dev`.

cathay4t commented:
Waiting on PR #228

EdDev commented:
I can't see any 'Adding Fedora CI docker image.' related commit in the PR. Looks like it was missed and only the dependent PR/s commits are here.

cathay4t commented:
The Centos docker file changes will be in a different PR.

cathay4t commented:
Wait on PR #229 in order to have a stable CI test feedback.

EdDev commented:
I must say it makes be a bit worried to see these scripts extracted, just because someone may by mistake run them not within a container (and ruin his env).
Perhaps we should make sure it is not saved as an executable in the repo.

tyll commented:
> I must say it makes be a bit worried to see these scripts extracted, just because someone may by mistake run them not within a container (and ruin his env).
> Perhaps we should make sure it is not saved as an executable in the repo.

What if someone accidentally runs nmstate and ruin their env? ;-) - the scripts still need root privileges to hurt but I guess not making them executable, removing the shebang and calling them directly with bash does not work.

EdDev commented:
Depending on how docker is installed, developers run now `sudo ./automation/run-integration-tests.sh`, I think that it is not that imaginary to wrongly run a different script instead.
For running nmstate and ruin your setup you need to install it, then know the exact syntax to ruin your config. Here it is one liner away from that.

Can't we convert it into an executable in the container itself?

tyll commented on automation/docker_enable_systemd.sh:
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Since the scripts are called with bash ..., the shebang lines can be removed.

tyll commented on automation/docker_sys_config.sh:
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Since the scripts are called with bash ..., the shebang lines can be removed.

cathay4t commented:
V2 patch uploaded.

Change since V1:
 * Change the file mode of docker scripts to 0644 and removed the sha-bang.

Also waiting on other PR #235 to pass the CI.

tyll commented:
Thank you, it seems that you addressed all the open issues
